cvat-ui/src/containers/annotation-page/top-bar/top-bar.tsx

```diff
@@ -105,30 +130,84 @@ function mapDispatchToProps(dispatch: any): DispatchToProps {
     };
 }
 
-type Props = StateToProps & DispatchToProps;
+type Props = StateToProps & DispatchToProps & RouteComponentProps;
 class AnnotationTopBarContainer extends React.PureComponent<Props> {
+    private static beforeUnloadCallback(event: BeforeUnloadEvent): any {
+        const confirmationMessage = 'You have unsaved changes, please confirm leaving this page.';
+        // eslint-disable-next-line no-param-reassign
+        (event || window.event).returnValue = confirmationMessage;
+        return confirmationMessage;
+    }
+
+    private autoSaveInterval: number | undefined;
+    private unblock: any;
+
+    componentDidMount(): void {
+        const {
+            autoSave,
+            autoSaveInterval,
+            saving,
+        } = this.props;
+
+        this.autoSaveInterval = window.setInterval((): void => {
+            if (autoSave && !saving) {
+                this.onSaveAnnotation();
+            }
+        }, autoSaveInterval);
+
+        this.checkUnsavedChanges();
+    }
+
     public componentDidUpdate(): void {
         const {
             jobInstance,
             frameNumber,
+            frameDelay,
             playing,
             canvasIsReady,
-            onChangeFrame,
             onSwitchPlay,
         } = this.props;
 
+
         if (playing && canvasIsReady) {
             if (frameNumber < jobInstance.stopFrame) {
                 setTimeout(() => {
                     const { playing: stillPlaying } = this.props;
                     if (stillPlaying) {
-                        onChangeFrame(frameNumber + 1);
+                        this.onChangeFrame(frameNumber + 1);
                     }
-                });
+                }, frameDelay);
             } else {
                 onSwitchPlay(false);
             }
         }
+
+        this.checkUnsavedChanges();
+    }
+
+    public componentWillUnmount(): void {
+        window.clearInterval(this.autoSaveInterval);
+        window.removeEventListener('beforeunload', AnnotationTopBarContainer.beforeUnloadCallback);
+        if (typeof this.unblock === 'function') {
+            this.unblock();
+            this.unblock = undefined;
+        }
+    }
+
+    private onChangeFrame(newFrame: number): void {
+        const {
+            canvasInstance,
+            onChangeFrame,
+            resetZoom,
+        } = this.props;
+
+        onChangeFrame(newFrame);
+
+        if (resetZoom) {
+            canvasInstance.html().addEventListener('canvas.setup', () => {
+                canvasInstance.fit();
+            }, { once: true });
+        }
```

Comment:
I would prefer do it in a ``canvas-wrapper`` component. 
In this case we have one extra relation (top bar container knows about canvas).